File: dart/Polymorphism.dart
class Animal {
  void sound() {
    print('Animal sound');
  }
}

class Dog extends Animal {
  @override
  void sound() {
    print('Woof');
  }
}

class Cat extends Animal {
  @override
  void sound() {
    print('Meow');
  }
}

void main() {
  List<Animal> animals = [Dog(), Cat()];
  for (var animal in animals) {
    animal.sound();
  }
}

File: java/Polymorphism.java
class PolyAnimal {
    void sound() {
        System.out.println("Animal sound");
    }
}

class PolyDog extends PolyAnimal {
    @Override
    void sound() {
        System.out.println("Woof");
    }
}

class PolyCat extends PolyAnimal {
    @Override
    void sound() {
        System.out.println("Meow");
    }
}

public class Polymorphism {
    public static void main(String[] args) {
        PolyAnimal animal1 = new PolyDog();
        PolyAnimal animal2 = new PolyCat();
        animal1.sound();
        animal2.sound();
    }
}
